token/program/src/processor.rs

```diff
@@ -768,7 +770,8 @@ impl Processor {
         // make sure the mint is valid
         let mint_info = next_account_info(account_info_iter)?;
         Self::check_account_owner(program_id, mint_info)?;
-        let _ = Mint::unpack(&mint_info.data.borrow())?;
+        let _ = Mint::unpack(&mint_info.data.borrow())
+            .map_err(|_| Into::<ProgramError>::into(TokenError::InvalidMint))?;
```

Comment:
This change is a little out of the scope of this PR, but happened on it while comparing test outputs between the two token programs, which were a little inconsistent.